File: frontend/src/api/order.ts
import api from './index'

export interface Card {
  id: number
  content: string
  status: string
}

export interface Order {
  id: number
  order_no: string
  product_id: number
  product_name: string
  quantity: number
  unit_price: number
  total_price: number
  coupon_code: string
  discount: number
  status: string
  cards: Card[]
  created_at: string
}

export const orderApi = {
  create(data: { product_id: number; quantity: number; coupon_code?: string }) {
    return api.post<Order>('/orders', data)
  },

  detail(id: number) {
    return api.get<Order>(`/orders/${id}`)
  },

  myOrders() {
    return api.get<Order[]>('/orders/my')
  },
}

File: backend/internal/model/order.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Order struct {
	ID          uint           `json:"id" gorm:"primaryKey"`
	OrderNo     string         `json:"order_no" gorm:"uniqueIndex;not null"`
	UserID      uint           `json:"user_id" gorm:"not null"`
	ProductID   uint           `json:"product_id" gorm:"not null"`
	ProductName string         `json:"product_name" gorm:"not null"`
	Quantity    int            `json:"quantity" gorm:"not null"`
	UnitPrice   float64        `json:"unit_price" gorm:"not null"`
	TotalPrice  float64        `json:"total_price" gorm:"not null"`
	CouponCode  string         `json:"coupon_code"`
	Discount    float64        `json:"discount" gorm:"default:0"`
	Status      string         `json:"status" gorm:"default:pending"`
	Cards       []Card         `json:"cards" gorm:"foreignKey:OrderID"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`
}
